clients/python/dd_image_client.py

```diff
@@ -0,0 +1,316 @@
+            image_vectors[img_path] = response['vals']
+            print("files left to process " + str(len(image_files)))
+
+        target_file = open(target_file_path, 'w')
+        json.dump(image_vectors, target_file)
+        target_file.close()
+
+    def classify_images_from_dir(self, dir_path):
+
+        image_files = self._fetch_files_from_dir(dir_path)
+
+        while (len(image_files)) > 0:
+            image_file_path = image_files.pop()
+            img_data = self._pre_process_image(image_file_path)
+
+            classes = self._make_image_classification_call(img_data)
+            if classes is not None:
+                print(image_file_path + ': ' + classes['cat'] + ' with probability ' + str(classes['prob']))
+            else:
+                print("error making classification call for " + image_file_path)
+
+    def delete_service(self):
+        self.dd.delete_service(self.sname, clear='')
+
+    def _make_image_classification_call(self, img_data):
+        parameters_input = {}
+        parameters_mllib = {}
+        parameters_output = {'best': 1}
+        prediction_call = None
+        # noinspection PyBroadException
+        try:
+            t0 = time.time()
+            prediction_call = self.dd.post_predict(self.sname, img_data, parameters_input, parameters_mllib,
+                                                   parameters_output)
+            classes = prediction_call['body']['predictions']['classes']
+            t1 = time.time()
+            print ("received response in " + str(t1 - t0) + " seconds")
+
+            return classes
+        except Exception:
+            print("Error in response!! " + str(prediction_call['status']))
+            return None
+
+    def _make_feature_extraction_call(self, image_data):
+        parameters_input = {}
+        parameters_mllib = {'gpu': self.GPU, 'extract_layer': self.extract_layer}
+        parameters_output = {'binarized': self.binarized}
+        predict_call = None
+        try:
+            t0 = time.time()
+            predict_call = self.dd.post_predict(self.sname, image_data, parameters_input, parameters_mllib,
+                                                parameters_output)
+            response = predict_call['body']['predictions']
+            t1 = time.time()
+            print ("received responses in " + str(t1 - t0) + " seconds")
+            return response
+        except Exception:
+            if predict_call is not None:
+                print("Error in server response " + str(predict_call['status']))
+            else:
+                print("Error in server response")
+        return None
+
+    def _convert_img_to_base64(self, img):
+        tmp_buffer = cStringIO.StringIO()
+        img.save(tmp_buffer, format="JPEG")
+        base64_img_str = base64.b64encode(tmp_buffer.getvalue())
+        tmp_buffer.close()
+        return base64_img_str
+
+    def _rotate_image_if_needed(self, img, path):
```

Comment:
need to give you some context here. I implemented this mainly because I am working with mobile images which for the most part come rotated horizontally. I agree this doesnt belong in the client.